lms/static/js/edxnotes/plugins/llm_summarize.js

```diff
@@ -68,17 +68,113 @@
                         background-color: white;
                         border: 1px solid gray;
                     }
+
+                    .summarize-loader-wrapper {
+                        display: none;
+                    }
+
+                    .summarize-loader-wrapper.show {
+                        display: block
+                    }
+                    // Defining content loader since fontawesome icons don't work
+                    // inside the annotatorjs modal.
+                    .loader {
+                        width: 5em !important;
+                        border-color: red !important;
+                    }
+                    @keyframes rotation {
+                    0% {
+                        transform: rotate(0deg);
+                    }
+                    100% {
+                        transform: rotate(360deg);
+                    }
                 `;
                 document.head.appendChild(style);
-                var annotator = this.annotator;
+                this.modifyDom(this.annotator);
+                const summarizeButton = document.getElementById('summarizeButton');
+                summarizeButton.addEventListener('click', function(ev) {
+                    document.querySelector('.annotator-outer.annotator-editor').setAttribute('is_summarizing', true);
+                });
+                this.annotator.subscribe('annotationEditorShown', this.handleSummarize);
+                this.annotator.subscribe('annotationEditorHidden', this.cleanupSummarize);
+            },
+            handleSummarize: function (editor, annotation) {
+                if (editor.element[0].getAttribute('is_summarizing') !== 'true') return;
+                const textAreaWrapper = editor.fields[0].element;
+                editor.fields[1].element.children[0].value = 'ai_summary';
+                toggleLoader(editor);
+
+                const request = new Request("/pearson-core/llm-assistance/api/v0/summarize-text", {
```

Comment:
Please, do not mix single and double quotes. @alexjmpb 
Where does Request class come from?

Reply:
It is built-in https://developer.mozilla.org/en-US/docs/Web/API/Request